Lecture 5

HidJaba-hub opened the pull request.

HidJaba-hub committed:
Fixed issues:
-fixed viewedProducts checking for emptiness
-removed .super overriding
-renamed getOrder
-added default values
-added tests
-added too many request jsp
-added exceptions
-added abstract class

HidJaba-hub committed:
Lecture 5

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/DAO/orderDao/OrderDao.java:
```diff
@@ -0,0 +1,13 @@
+
+    Optional<Order> getOrderBySecureId(String id) throws OrderNotFoundException;
+
+    void save(Order order) throws OrderNotFoundException;
```
where OrderNotFoundException is thrown?

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/service/order/CustomOrderService.java:
```diff
@@ -0,0 +1,79 @@
+
+    @Override
+    public void placeOrder(Order order) {
+        order.setSecureId(UUID.randomUUID().toString());
```
id setting should not happen in this sevice

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/service/order/OrderService.java:
```diff
@@ -0,0 +1,18 @@
+
+    Order getOrder(Cart cart);
+
+    Order getOrderById(String id);
```
exceptions?

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/web/filter/DosFilter.java:
```diff
@@ -0,0 +1,37 @@
+        if (dosProtectionService.isAllowed(servletRequest.getRemoteAddr())) {
+            filterChain.doFilter(servletRequest, servletResponse);
+        } else {
+            ((HttpServletResponse) servletResponse).setStatus(429);
```
reserved constant can be used

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/validators/DefaultParseValidator.java:
```diff
@@ -0,0 +1,51 @@
+
+    @Override
+    public void validateName(String stringToValidate, Map<String, String> errors, String parameter) {
+        if (!stringToValidate.matches("^[a-zA-Z][\\sa-zA-Z]*$")) {
```
russian letters are failing with  Remove numbers, please

esSoftPalinaTsi commented:
Recently viewed title can be protected by a check if this list is not empty
If there's no product, this title now is always present

HidJaba-hub committed:
Fixed issues:
-added tests
-added too many request jsp
-added exceptions

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/security/CustomDosProtectionService.java:
```diff
@@ -1,15 +1,18 @@
 public class CustomDosProtectionService implements DosProtectionService {
 
-    private static final long TIME_INTERVAL_MILLS = 60000;
+    private static final long TIME_INTERVAL_MILLS = 60_000;
```
why 60_000

HidJaba-hub replied:
this is for clarity. "_" not taken into account when compiling

esSoftPalinaTsi commented:
can we add phone format to the input? to make it visible to customer before he submits the form

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/DAO/orderDao/CustomOrderDao.java:
```diff
@@ -0,0 +1,42 @@
+
+    @Override
+    public void save(Order order) throws IllegalArgumentException {
+        super.save(order);
```
if we call inly super, then this method can be deleted from dao at all
and only parent method will be called by default

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/service/order/CustomOrderService.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    @Override
+    public Order getOrder(Cart cart) {
```
createOrder?

HidJaba-hub committed:
Fixed issues:
-fixed viewedProducts checking for emptiness
-removed .super overriding
-renamed getOrder
-added default values

esSoftPalinaTsi commented on src/main/java/com/es/phoneshop/DAO/GenericDao.java:
```diff
@@ -0,0 +1,76 @@
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
+
+public class GenericDao<T extends GenericEntity> {
```
should be abstract class

HidJaba-hub committed:
-added abstract class

HidJaba-hub committed:
Merge remote-tracking branch 'origin/lecture-5' into lecture-5

esSoftPalinaTsi commented:
Add stock decrease after the placing order

HidJaba-hub committed:
Fixed issues:
-added stock recalculate
-added tests
-added "Product is out of stock" if stock below zero

HidJaba-hub committed:
Fixed issues:
-added stock recalculate
-added tests
-added "Product is out of stock" if stock below zero
-removed other default values

esSoftPalinaTsi commented on src/main/webapp/WEB-INF/pages/checkout.jsp:
```diff
@@ -0,0 +1,119 @@
+    <h2>Your details</h2>
+    <table>
+    <table>
+        <tags:orderFormRow name="firstName" label="First name" order="${order}" errors="${errors}" type="text" defaultValue="Ваше имя"/>
+        <tags:orderFormRow name="lastName" label="Last name" order="${order}" errors="${errors}" type="text" defaultValue="Ваша фамилия"/>
+        <tags:orderFormRow name="phone" label="Phone" order="${order}" errors="${errors}" type="text" defaultValue="+375(xx)xxxxxxx"/>
+        <tags:orderFormRow name="deliveryDate" label="Delivery date" order="${order}" errors="${errors}" type="date" defaultValue=""/>
+        <tags:orderFormRow name="deliveryAddress" label="Delivery address" order="${order}" errors="${errors}"
+                           type="text" defaultValue="г. Город ул.Ваша улица, дом"/>
```
let's leave default value only for phone
because for phone it's impossible to save your +375(xx)xxxxxxx due to validation
but for other fields we can save address, name and surname without filling values, just with the default ones

HidJaba-hub committed:
-removed other default values

HidJaba-hub committed:
Merge remote-tracking branch 'origin/lecture-5' into lecture-5

HidJaba-hub committed:
-removed other default values
